docs/architecture/provisioning-custom-configuration.md

```diff
@@ -0,0 +1,148 @@
+# Custom configuration of provisioned Runtimes
+
+## Introduction
+
+There is a need for configuring Kyma Runtimes provisioned by the Runtime Provisioner.
+For now only option is to configure Runtime manually after provisioning is finished and Kubeconfig is fetched.
+
+
+## Solution
+
+The proposed solution is to provide ability to provide the configuration for components, which then will be used as Installer overrides.
+
+Changes to the GraphQL schema:
+```graphql
+type Configuration {
+    key: String!
+    value: String!
+    secret: Boolean
+} 
+
+input ConfigurationInput {
+    key: String!
+    value: String!
+    secret: Boolean
+}
+
+type ComponentConfiguration {
+    component: KymaComponent
+    configuration: [Configuration]
+}
+
+input ComponentConfigurationInput {
+    component: KymaComponent!
+    configuration: [ConfigurationInput]
+}
+
+type KymaConfig {
```

Comment:
why do you call it KymaConfig and not just installation? it sounds strange if configuration passed inside is optional

what is this `KymaComponent` type all about, is it going to be enum or?

and to both above, do we really need this Kyma prefix?

Reply:
It is part of the already existing API. I do not see the reason why shouldn't it be called `KymaConfig`. `KymaComponent` is also the part of already existing API and it is an enum, currently called `KymaModule`, but as we agree we are changing it to `KymaComponent`.

The scope of the proposal is only extending the API with providing configuration, not the whole API design, which is partially already in place and implemented.